File: vent-soft/src/main/java/com/qualitysales/ventsoft/Controllers/DTO/InvoiceDTO.java
package com.qualitysales.ventsoft.Controllers.DTO;

import com.qualitysales.ventsoft.model.Client;
import com.qualitysales.ventsoft.model.ItemInvoice;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;

@AllArgsConstructor
@NoArgsConstructor
@Builder
@Data
public class InvoiceDTO {
    private Integer id;
    private String invoiceCode;
    private Client client;
    private String date;
    private BigDecimal total;
    private ItemInvoice itemInvoice;
    private String status;
}

File: vent-soft-front/src/app/business/invoices/invoice/invoice.dto.ts
import { CustomerDTO } from "../../customer/customer.dto";
import { ItemInvoiceDTO } from "./itemInvoice.dto";

export interface InvoiceDTO {
  id: number;
  invoiceCode: string;
  client: CustomerDTO;
  nit: string;
  email: string;
  cellPhone: string;
  address: string;
  date: string;
  total: number;
  itemInvoices: ItemInvoiceDTO[];
  status: boolean;
}

